adagrams/game.py

```diff
@@ -1,11 +1,110 @@
+                "L", "L", "L", "L", 
+                "M", "M",
+                "N", "N", "N", "N", "N", "N", 
+                "O", "O", "O", "O", "O", "O", "O", "O", 
+                "P", "P",
+                "Q",
+                "R", "R", "R", "R", "R", "R", 
+                "S", "S", "S", "S", 
+                "T", "T", "T", "T", "T", "T", 
+                "U", "U", "U", "U", 
+                "V", "V",
+                "W", "W",
+                "X",
+                "Y", "Y",
+                "Z"
+    ]
+
+
 def draw_letters():
-    pass
+    # create a copy of letter pool so we don't alter original list
+    letter_pool_copy = deepcopy(LETTER_POOL)
+    drawn_letters = []
+    
+    # until the length of drawn_letters is 10, keep generating
+    # a random index to pop off from the letter pool 
+    # and append popped element to drawn_letters
+    while len(drawn_letters) < 10:
+        rand_index = randint(0, len(letter_pool_copy)-1)
+        drawn_letters.append(letter_pool_copy.pop(rand_index))
+    return drawn_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # create a copy of letter bank so we don't alter original list
+    letter_bank_copy = deepcopy(letter_bank)
+    word = word.upper()
+    valid_word = True
+    # check all letters in word are available in correct quantity
+    for letter in word:
+        if letter not in letter_bank_copy:
+            valid_word = False
+        else:
+            letter_bank_copy.remove(letter)
+    
+    return valid_word
 
 def score_word(word):
-    pass
+    
+    letter_dict = {
+    'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 
+    'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+    'D': 2, 'G': 2,
+    'B': 3, 'C': 3, 'M': 3, 'P': 3,
+    'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+    'K': 5,
+    'J': 8, 'X': 8,
+    'Q': 10, 'Z': 10,
+}
+
+    word_score = 0
+    
+    if word == None:
+        return None
+    else:
+        word = word.upper()
+        for letter in word:
+            word_score += letter_dict[letter]
+        if len(word) >= 7:
+            word_score += 8
+    return word_score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+   
+    best_word = {
+        "word" : "",
+        "score" : 0
+    }
+
+    for word in word_list:
+        score = score_word(word)
+        if score > best_word["score"]:
+            update_best_word(best_word, word, score)
+        # check for ties using word length guidelines
+        elif score == best_word["score"]:
+            if len(word) == 10 and len(best_word["word"]) != 10:
+                update_best_word(best_word, word, score)
+            elif len(word) < len(best_word["word"]) and len(best_word["word"]) != 10:
+                update_best_word(best_word, word, score)
+
+    return (best_word["word"], best_word["score"])
+
+def update_best_word(best_word, word, score):
+    best_word["score"] = score
+    best_word["word"] = word
+    return best_word
```

Comment:
Python (and most languages, if I think of it) likes it when you have a newline at the end of the file. (Why? Because it's one of those [Just Unix/Computer Things](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file), I guess?)